common/persistence/sql/history_store.go

```diff
@@ -425,14 +426,72 @@ func (m *sqlExecutionStore) DeleteHistoryBranch(
 	})
 }
 
+// getAllHistoryTreeBranchesPaginationToken represents the primary key of the latest row in the history_tree table that
+// we returned.
+type getAllHistoryTreeBranchesPaginationToken struct {
+	ShardID  int32
+	TreeID   primitives.UUID
+	BranchID primitives.UUID
+}
+
 func (m *sqlExecutionStore) GetAllHistoryTreeBranches(
-	_ context.Context,
+	ctx context.Context,
 	request *p.GetAllHistoryTreeBranchesRequest,
 ) (*p.InternalGetAllHistoryTreeBranchesResponse, error) {
+	pageSize := request.PageSize
+	if pageSize <= 0 {
+		return nil, fmt.Errorf("PageSize must be greater than 0, but was %d", pageSize)
+	}
+
+	page := sqlplugin.HistoryTreeBranchPage{
+		Limit: pageSize,
+	}
+	if len(request.NextPageToken) != 0 {
+		var token getAllHistoryTreeBranchesPaginationToken
+		if err := json.Unmarshal(request.NextPageToken, &token); err != nil {
```

Comment:
it doesn't really matter since this is not getting exposed internally, but I thought we like to use proto encoding for tokens

Reply:
JSON is used for at least some internal pagination tokens like with `deserializeHistoryNodePaginationToken`